prov/rxd/src/rxd_cq.c

```diff
@@ -1101,6 +1114,22 @@ void rxd_handle_recv_comp(struct rxd_ep *ep, struct fi_cq_msg_entry *comp)
 	}
 }
 
+void rxd_handle_error(struct rxd_ep *ep)
+{
+	struct fi_cq_err_entry err = {0};
+	int ret;
+
+	ret = fi_cq_readerr(ep->dg_cq, &err, 0);
+	if (ret < 0) {
+		FI_WARN(&rxd_prov, FI_LOG_CQ,
+			"Error reading CQ: %s\n", fi_strerror(-ret));
+	} else {
+		FI_WARN(&rxd_prov, FI_LOG_CQ,
+			"Received %s error from core provider: %s\n",
+			err.flags & FI_SEND ? "tx" : "rx", fi_strerror(-err.err)); 
+	}
```

Comment:
You should probably fail the transfer for send failures.  But this shouldn't happen either.

Reply:
Yeah this is mainly there as a precaution for warning of any failures (which I added in trying to find that verbs memory corruption bug wondering if maybe the transfer completed in an error but I just didn't catch it). I can't see it ever getting here but figured to add it just in case.
As for failing the transfer - I considered that as well but it gets a bit trickier for larger sends with a lot of packets (ie, sequence numbers, do we clear the future packets because the message already failed, do we have to wait for completions per our in_use check discussion). All the extra implementation/handling of this didn't seem worth it to me for a situation that shouldn't happen. But I can add all that in if you prefer.